src/dag_algo/mod.rs

```diff
@@ -338,37 +338,50 @@ pub fn layers(
 /// topologically sorted using the provided key function. A topological sort
 /// is a linear ordering of vertices such that for every directed edge from
 /// node :math:`u` to node :math:`v`, :math:`u` comes before :math:`v`
-/// in the ordering.
+/// in the ordering.  If ``reverse`` is set to ``False``, the edges are treated
+/// as if they pointed in the opposite direction.
 ///
 /// This function differs from :func:`~rustworkx.topological_sort` because
 /// when there are ties between nodes in the sort order this function will
 /// use the string returned by the ``key`` argument to determine the output
-/// order used.
+/// order used.  The ``reverse`` argument does not affect the ordering of keys
+/// from this function, only the edges of the graph.
 ///
 /// :param PyDiGraph dag: The DAG to get the topological sorted nodes from
 /// :param callable key: key is a python function or other callable that
 ///     gets passed a single argument the node data from the graph and is
 ///     expected to return a string which will be used for resolving ties
 ///     in the sorting order.
+/// :param bool reverse: If ``False`` (the default), perform a regular
+///     topological ordering.  If ``True``, return the lexicographical
+///     topological order that would have been found if all the edges in the
+///     graph were reversed.  This does not affect the comparisons from the
+///     ``key``.
 ///
 /// :returns: A list of node's data lexicographically topologically sorted.
 /// :rtype: list
 #[pyfunction]
-#[pyo3(text_signature = "(dag, key, /)")]
+#[pyo3(signature = (dag, /, key, *, reverse=false))]
 pub fn lexicographical_topological_sort(
     py: Python,
     dag: &digraph::PyDiGraph,
     key: PyObject,
+    reverse: bool,
 ) -> PyResult<PyObject> {
     let key_callable = |a: &PyObject| -> PyResult<PyObject> {
         let res = key.call1(py, (a,))?;
         Ok(res.to_object(py))
     };
     // HashMap of node_index indegree
     let node_count = dag.node_count();
+    let (in_dir, out_dir) = if reverse {
+        (petgraph::Direction::Outgoing, petgraph::Direction::Incoming)
+    } else {
+        (petgraph::Direction::Incoming, petgraph::Direction::Outgoing)
+    };
```

Comment:
Can you create a public function in `dag_algo` to encapsulate this logic and reuse it in the other files? It is a nitpick but just to avoid the kind of bugs where we flip it

Reply:
Done in 12a91ec.